aidb/engine/engine.py

```diff
@@ -42,3 +49,26 @@ def execute(self, query: str, **kwargs):
       raise e
     finally:
       self.__del__()
+
+  async def clear_ml_cache(self, service_name_list = None):
+    '''
+    Clear the cache and output table if the ML model has changed.
+    Delete the tables following the inference services' topological order to maintain integrity during deletion.
+    service_name_list: the name of all the changed services. A list of str or None.
+    If the service name list is not given, the output for all the services will be cleared.
+    '''
+    async with self._sql_engine.begin() as conn:
+      service_ordering = self._config.inference_topological_order
+      if service_name_list is None:
+        service_name_list = [bounded_service.service.name for bounded_service in service_ordering]
+      service_name_list = set(service_name_list)
+      for bounded_service in reversed(service_ordering):
+        if isinstance(bounded_service, CachedBoundInferenceService):
+          if bounded_service.service.name in service_name_list:
+            for input_column in bounded_service.binding.input_columns:
+              service_name_list.add(input_column.split('.')[0])
+            asyncio_run(conn.execute(delete(bounded_service._cache_table)))
+            for output_column in bounded_service.binding.output_columns:
+              asyncio_run(conn.execute(delete(bounded_service._tables[output_column.split('.')[0]]._table)))
```

Comment:
The output tables may be same. Can you add them into a set and then delete them?